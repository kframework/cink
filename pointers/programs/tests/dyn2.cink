int* f() {
  int *p = new int;
  *p = 10;
  return p;
}

int main() {
  int *a = f();
  cout << *a << endl;
  delete a;
  return 0;
} 
