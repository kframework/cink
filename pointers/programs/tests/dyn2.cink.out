10
<T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            0
        </return>
        <fstack>
            .List
        </fstack>
        <sideEffects>
            .Bag
        </sideEffects>
        <env>
            f |-> #symLoc(1)
            main |-> #symLoc(2)
        </env>
    </control>
    <genv>
        f |-> #symLoc(1)
        main |-> #symLoc(2)
    </genv>
    <store>
        #symLoc(1) |-> lambda ( .Decls @ ((((p of (pointer to int) ;) ((p 
          = (new int)) ;)) (((* p) = 10) ;)) (return p ;)) )
        #symLoc(2) |-> lambda ( .Decls @ (((((a of (pointer to int) ;) ((a = (
          f ( '.Exps(.KList) ))) ;)) (((cout << (* a)) << endl) ;)) ((delete a
          ) ;)) (return 0 ;)) )
        #symLoc(4) |-> #symLoc(5)
        #symLoc(5) |-> 10
    </store>
</T>
